### PR TITLE
Add Glue Catalog Explorer

### PR DESCRIPTION
**Pull request by lmouhib**
*Issue #, if available:*

*Description of changes:*
This PR added the ability to explore a Glue Data Catalog from VS Code.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/clients/glueClient.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+    public async listDatabases(): Promise<Database[]> {
+        this.globals.outputChannel.appendLine(
+            `Glue Catalog: Fetching databases from ${this.globals.awsContext.getRegion()} region.`
+        );
+        const glue = await this.createGlueClient();
+        let databases;
+        try {
+            // Note that this requires aws-sdk<=v3.30.0
+            // due to https://github.com/aws/aws-sdk-js-v3/issues/3511
```
Comment (dacort) [inline src/clients/glueClient.ts:34]: These lines can be removed.

```suggestion
```

File: src/clients/glueClient.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+        try {
+
+            let input: GetTablesRequest = {
+                // eslint-disable-next-line @typescript-eslint/naming-convention
+                DatabaseName: databaseName,
+            };
+
+            // Note that this requires aws-sdk<=v3.30.0
+            // due to https://github.com/aws/aws-sdk-js-v3/issues/3511
```
Comment (dacort) [inline src/clients/glueClient.ts:65]: Remove
```suggestion
```

File: src/clients/glueClient.ts [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        }
+
+        return tables;
+
+    }
+
+    public async getTable(tableName: string, databaseName: string): Promise<Table | undefined>{
+
+        this.globals.outputChannel.appendLine(
+            `Glue Catalog: Fetching tables of databse ${databaseName} from ${this.globals.awsContext.getRegion()} region.`
```
Comment (dacort) [inline src/clients/glueClient.ts:85]:
```suggestion
            `Glue Catalog: Fetching tables of database ${databaseName} from ${this.globals.awsContext.getRegion()} region.`
```

File: src/clients/glueClient.ts [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+            let input: GetTableRequest = {
+                // eslint-disable-next-line @typescript-eslint/naming-convention
+                DatabaseName: databaseName,
+                // eslint-disable-next-line @typescript-eslint/naming-convention
+                Name: tableName,
+            };
+
+            // Note that this requires aws-sdk<=v3.30.0
+            // due to https://github.com/aws/aws-sdk-js-v3/issues/3511
```
Comment (dacort) [inline src/clients/glueClient.ts:100]: Remove
```suggestion
```

File: src/explorer/glueCatalog.ts [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    );
+  }
+
+}
+
+export class GlueCatalogTable extends vscode.TreeItem {
+  constructor(
+    private readonly gluetable: Table
+  ) {
+    super(`${gluetable.Name} v:${gluetable.VersionId}`);
```
Comment (dacort) [inline src/explorer/glueCatalog.ts:78]: I find the `v:` a little distracting. Maybe we can tweak it a bit so it's shown as the description on the table and remove the `:`.

```suggestion
    super(`${gluetable.Name}`);
    this.description = `v${gluetable.VersionId}`;
```

File: src/extension.ts [hunk 6]
```diff
@@ -8,12 +8,16 @@ import { AwsContextCommands } from "./aws_context";
...
 import { DefaultEMRServerlessClient } from "./clients/emrServerlessClient";
+import { DefaultGlueClient } from "./clients/glueClient";
 import { EMREC2Filter } from "./emr_explorer";
 import { EMRLocalEnvironment } from "./emr_local";
 import { copyIdCommand } from "./explorer/commands";
 import { EMRContainersNode } from "./explorer/emrContainers";
 import { EMRNode } from "./explorer/emrEC2";
 import { EMRServerlessNode } from "./explorer/emrServerless";
+import { GlueCatalogNode } from "./explorer/glueCatalog";
+import { getWebviewContent, GlueTablePanel } from "./panels/glueTablePanel";
```
Comment (dacort) [inline src/extension.ts:19]: I don't think `GlueTablePanel` is used.
```suggestion
import { getWebviewContent } from "./panels/glueTablePanel";
```

**Review by dacort — CHANGES_REQUESTED** [review]
Looks pretty good and works as expected. A few minor changes.

One other thought - there's no default styling so it's a little tough to read the table details. It took a while, but I figured out how to add some CSS so may spruce it up a bit once merged. :)

File: src/explorer/glueCatalog.ts (hunk 5 above)
Comment (lmouhib) [reply src/explorer/glueCatalog.ts:78]: Ok, we can in future release a version explorer in the web view. For now we always display the latest.

File: src/extension.ts (hunk 6 above)
Comment (lmouhib) [reply src/extension.ts:19]: Agree, that's left from one of my experiments with web views.

**Review by dacort — APPROVED** [review]

**Comment by lmouhib** [comment]
> One other thing I noticed is that `struct` data types get interpreted as HTML (with the `<`). I'll figure out that change before I merge.

Ow, Interesting, I have not seen this, maybe an escape character? it would probably mean we need to do some check before we pass the jsScript.

**Comment by dacort** [comment]
Yea, just need to change `innerHTML` to `innerText`.

**Pull request merged**